### ISSUE TITLE
Flye keeps erroring out at plasmid step 

### ISSUE THREAD

**Issue by sxh1136** [fenderglass/ABruijn#102]
I've tried reinstalling conda, creating new conda environments and reinstalling flye multiple times but it still errors out. 
This is the command I'm running:
flye --nano-raw <input fastq> --genome-size 240m --out-dir <directory> --threads 64 --meta --plasmids

This is the traceback:
File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/managers.py", line 742, in _connect
    conn = self._Client(self._token.address, authkey=self._authkey)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 169, in Client
    c = SocketClient(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 308, in SocketClient
    s.connect(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 2] No such file or directory
Process Process-91:
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/site-packages/flye/polishing/bubbles.py", line 80, in _thread_worker
    error_queue.put(e)
  File "<string>", line 2, in put
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/managers.py", line 755, in _callmethod
    self._connect()
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/managers.py", line 742, in _connect
    conn = self._Client(self._token.address, authkey=self._authkey)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 169, in Client
    c = SocketClient(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 308, in SocketClient
    s.connect(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 2] No such file or directory
Process Process-102:
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/process.py", line 267, in _bootstrap
    self.run()
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/site-packages/flye/polishing/bubbles.py", line 80, in _thread_worker
    error_queue.put(e)
  File "<string>", line 2, in put
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/managers.py", line 755, in _callmethod
    self._connect()
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/managers.py", line 742, in _connect
    conn = self._Client(self._token.address, authkey=self._authkey)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 169, in Client
    c = SocketClient(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/multiprocessing/connection.py", line 308, in SocketClient
    s.connect(address)
  File "/home/ubuntu/miniconda3/envs/flye/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 2] No such file or directory

**Comment by mikolmogorov** [comment]
Hi,

Can you post the `flye.log` file?

**Comment by sxh1136** [comment]
[flye.log](https://github.com/fenderglass/Flye/files/3099980/flye.log)

Thanks in advance.

**Comment by mikolmogorov** [comment]
Thank you. 

Something seems to be happening during plasmid polishing, but the python backtrace is not very informative.. A couple suggestions:

1. Can you post the output of `ls -l flye_dir/22-plasmids`?

2. Try to skip the plasmid assembly stage (run without `--plasmids`). You can resume the current run with `--resume-from contigger` to save time.This way you should get your assembly, and we will see if the problem is related to the plasmids stage only.

**Comment by sxh1136** [comment]
This is the ls -l output:

```
total 38693684
-rw-rw-r-- 1 ubuntu ubuntu           0 Apr 19 23:01 bubbles_1.fasta
-rw-rw-r-- 1 ubuntu ubuntu      975854 Apr 19 22:56 chunks_1.fasta
-rw-rw-r-- 1 ubuntu ubuntu 11665621795 Apr 19 23:01 minimap_1.sam
-rw-rw-r-- 1 ubuntu ubuntu      973958 Apr 19 22:56 plasmids_raw.fasta
-rw-rw-r-- 1 ubuntu ubuntu   960293404 Apr 19 20:46 reads2contigs.paf
-rw-rw-r-- 1 ubuntu ubuntu     4589090 Apr 19 22:56 trimmed.paf
-rw-rw-r-- 1 ubuntu ubuntu     8803584 Apr 19 22:56 trimmed_sequences.fasta
-rw-rw-r-- 1 ubuntu ubuntu 15431615334 Apr 19 22:05 unmapped_ava.paf
-rw-rw-r-- 1 ubuntu ubuntu 11549441103 Apr 19 20:53 unmapped_reads.fasta
```
Thanks I'll try to resume it from contigger
>

**Comment by sxh1136** [comment]
Assembler finished with no problems when resuming from contigger 🤔

**Comment by mikolmogorov** [comment]
Thank you. Looks like the issue is specific to the polishing during plasmid generation step. I'll take a look into this - could you also tell me what Flye version were you using?

**Comment by sxh1136** [comment]
Its the newest one 2.4.2

**Comment by semarpetrus** [comment]
Hi,
I am having a similar issue. Would setting "-i" to 0 prevent the polishing step? and what ramifications would that have on the assembly?

**Comment by mikolmogorov** [comment]
@semarpetrus did it also happen during the plasmid step? If so, if you just remove the `--plasmids` option, it should go through.

Could you also send me the log file and the output of `ls -l flye_dir/22-plasmids`?

**Comment by semarpetrus** [comment]
It seems like it was during the plasmid step; the step never finished. When I remove the `--plasmids` option it works fine.
The contents of 22-plasmids are:
total 56570965
-rw-rw-r-- 1 spetrus audio    33371655 Jun 27 08:43 reads2contigs.paf
-rw-rw-r-- 1 spetrus audio 51068415517 Jun 27 08:50 unmapped_ava.paf
-rw-rw-r-- 1 spetrus audio   267682833 Jun 27 08:44 unmapped_reads.fasta
and the log file: [flye.log](https://github.com/fenderglass/Flye/files/3336898/flye.log)

Could the genome size parameter be affecting this? I am working with metagenomic samples and it's difficult to estimate the genome size. 

I would really like to utilize this feature.

**Comment by mikolmogorov** [comment]
@semarpetrus thank you!

Seems like this crash even happened at a different part of the algorithm. Is it possible for you to send me your data? That would be the fastest way to fix the issue. Feel free to reach me at fenderglass@gmail.com if so.

**Comment by mikolmogorov** [comment]
@sxh1136  and @semarpetrus 

I think I found the issue that likely led to you crashes. In some cases, if a lot of reads were not assembled (which is often the case for metagenomes), the Python code was consuming a lot of RAM (way more than expected).

I pushed an update into the `flye` branch which should fix this behavior - please check if it works for you. You can restart the pipeline using `--resume-from plasmids` option (the other options such as `--meta` and `--plasmids` should also be added as in the original run).

**Comment by mikolmogorov** [comment]
Seems that the crashes during the plasmid assembly should be resolved in the new 2.5 release. Feel free to reopen the issue if the problem is still happening.
